tests/posts.test.js

```diff
@@ -0,0 +1,157 @@
+const request = require("supertest");
+const initApp = require("../server.js");
+const mongoose = require("mongoose");
+const postModel = require("../models/posts_model.js");
+
+let app;
+beforeAll(async () => {
+  app = await initApp();
+});
+beforeEach(async () => {
+  await postModel.deleteMany();
+});
+
+afterAll(async () => {
+  mongoose.connection.close();
```

Comment:
when there are no posts?
when post is not found
when post_id is invalid
post is not found
when parameter is missing